src/app/core/bootstrap/startup.service.ts

```diff
@@ -23,10 +22,10 @@ export class StartupService {
   load() {
     return new Promise((resolve, reject) => {
       this.authService
-        .onChange()
+        .user()
```

Comment:
Will it be better to set a generics type?

Reply:
Any suggestions?
I will change it to <User|null>

Reply:
`this.authService.user<User>()`

If we can set user method a generics type, the developer don't need to modify the auth service.

Reply:
Could you give me a example ?

Reply:
```
user<T=User>(): Observable<T> {
  return this.user$.pipe(share());
}

this.authService.user<CustomUser>()
```

In a real app, the user type maybe different.

Reply:
user() can return any?

Reply:
like this https://stackoverflow.com/questions/49622045/in-typescript-what-does-t-mean